src/jit/lower.cpp

```diff
@@ -788,6 +792,80 @@ GenTreePtr Lowering::NewPutArg(GenTreeCall* call, GenTreePtr arg, fgArgTabEntryP
     isOnStack = info->regNum == REG_STK;
 #endif // !FEATURE_UNIX_AMD64_STRUCT_PASSING
 
+#ifdef _TARGET_ARM_
+    // Strcut can be splitted into register(s) and stack in ARM
+    if (info->isSplit)
+    {
+        if (arg->OperGet() == GT_FIELD_LIST)
+        {
+            GenTreeFieldList* fieldListPtr       = arg->AsFieldList();
+            GenTreeFieldList* regFieldListHeader = fieldListPtr;
+            GenTreeFieldList* stkFieldListHeader = nullptr;
+            GenTreeFieldList* prevPtr            = nullptr;
+            GenTreePtr        newOper            = nullptr;
+
+            assert(fieldListPtr->IsFieldListHead());
+
+            for (unsigned ctr = 0; fieldListPtr != nullptr; fieldListPtr = fieldListPtr->Rest(), ctr++)
+            {
+                GenTreePtr curOp  = fieldListPtr->gtOp.gtOp1;
+                var_types  curTyp = curOp->TypeGet();
+
+                if (info->numRegs == ctr)
+                {
+                    // TODO: Need to check correctness for FastTailCall
+                    if (call->IsFastTailCall())
+                    {
+                        NYI_ARM("lower: struct argument by fast tail call");
+                    }
+
+                    // Split FieldList: Two FieldLists for registers and stack
```

Comment:
Wouldn't it make sense to leave the field list as-is, and add a flag to `GenTreeFieldList` to indicate whether it goes on the stack or in a register?